pyccel/ast/headers.py

```diff
@@ -587,24 +587,52 @@ def funcs(self):
 
 #==============================================================================
 class MacroFunction(Header):
-    """."""
+    """Represents a macro function header in the code.
+    A macro function header maps a defined function name, set of arguments
+    and results to a different function call. This is notably useful when
+    mapping python functions to fortran functions which do the same thing but
+    do not have the same set of arguments/results.
+
+    Parameters
+    ----------
+
+    name: str
+        the name used to call the function
+
+    args: iterable
+        The python arguments of the macro
+
+    master: str
+        The name of the function in Fortran
+
+    master_args: iterable
+        The Fortran arguments of the macro
+
+    results: iterable
+        The python results of the macro
+
+    results_shapes: iterable
+        A list of shapes of the results
+    """
+
     __slots__ = ('_name','_arguments','_master','_master_arguments',
-                 '_results','_copies_required')
+                 '_results','_copies_required', '_results_shapes')
 
-    def __init__(self, name, args, master, master_args, results=None):
+    def __init__(self, name, args, master, master_args, results=None, results_shapes=None):
         if not isinstance(name, str):
             raise TypeError('name must be of type str or PyccelSymbol')
 
         # master can be a string or FunctionDef
         if not isinstance(master, (str, FunctionDef, Interface)):
             raise ValueError('Expecting a master name of FunctionDef')
 
-        self._name             = name
-        self._arguments        = args
-        self._master           = master
-        self._master_arguments = master_args
-        self._results          = results
-        self._copies_required  = [a in self._results for a in self._arguments]
+        self._name              = name
+        self._arguments         = args
+        self._master            = master
+        self._master_arguments  = master_args
+        self._results           = results
+        self._results_shapes        = results_shapes
```

Comment:
For some reason this line is aligned differently to the rest. Please could you fix it

Reply:
Fixed d21b82e